tests/schedulers/test_direct.py

```diff
@@ -70,3 +70,37 @@ def test_submit_script_with_num_cores_per_mpiproc(scheduler, template):
     )
     result = scheduler.get_submit_script(template)
     assert f'export OMP_NUM_THREADS={num_cores_per_mpiproc}' in result
+
+
+@pytest.mark.timeout(timeout=10)
+def test_kill_job(scheduler):
+    """Test if kill_job kill all descendant children from the process.
+    For that we spawn a new process that runs a sleep command, then we
+    kill it and check if the sleep process is still alive.
+
+    current process           forked process            run command
+         python───────────────────python───────────────────sleep
+                           we kill this process    we check if still running
+    """
+    import multiprocessing
+
+    from aiida.transports.plugins.local import LocalTransport
+    from psutil import Process
+
+    def run_sleep_100():
+        import subprocess
+
+        subprocess.run(['sleep', '100'], check=False)
+
+    forked_process = multiprocessing.Process(target=run_sleep_100)
+    forked_process.start()
+    while len(forked_process_children := Process(forked_process.pid).children()) != 1:
+        pass
+    assert len(forked_process_children) == 1
+    sleep_process = forked_process_children[0]
+    with LocalTransport() as transport:
+        scheduler.set_transport(transport)
+        scheduler.kill_job(forked_process.pid)
+    while sleep_process.is_running():
```

Comment:
Thanks a lot @agoscinski , insightful tests!

Just a minor point, here only checks for immediate children, not the whole decedent tree. 
Can you maybe also add a children of children, just to check that the entire tree is getting killed(?)

Reply:
<img src="https://github.com/user-attachments/assets/b11e4089-adaf-4f9d-98fb-3167f16b204a" alt="jack" width="300"/>

LGTM!